demos/common/python/openvino/model_zoo/model_api/adapters/model_adapter.py

```diff
@@ -16,17 +16,27 @@
 
 import abc
 from dataclasses import dataclass, field
-from typing import Dict, List
+from typing import Dict, List, Set
 
 
 @dataclass
-class Metadata:
+class LayerMetadata:
+    names: Set[str] = field(default_factory=set)
     shape: List[int] = field(default_factory=list)
+    layout: str = ''
     precision: str = ''
     type: str = ''
     meta: Dict = field(default_factory=dict)
 
 
+def get_layout_from_shape(shape):
+    if len(shape) != 4:
+        raise RuntimeError('Get_layout supports only 4D input shape')
+
+    layout = 'NCHW' if shape[1] == 3 else 'NHWC'
```

Comment:
This will not give the correct result for models for 1-channel input images (for example medical imaging models).